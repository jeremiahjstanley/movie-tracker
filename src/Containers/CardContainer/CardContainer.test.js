import React from 'react';
import { CardContainer } from './index.js';
import { shallow, mount } from 'enzyme';


describe('CardContainer', () => {
  it('should match the snapshot', () => {
    const mockMovies = [{title:'ConAir', id:7, poster_path:'google.com'}];
    const mockFavorites = [{title:'ConAir', movie_id:9}];
    const wrapper = shallow(<CardContainer movies={mockMovies} favorites={mockFavorites} />, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke checkFavorites when favorite button is clicked', () => {
    const mockMovies = [{ title: 'ConAir', id: 7, poster_path: 'google.com' }];
    const mockFavorites = [{ title: 'ConAir', movie_id: 9 }];
    const mockCheckFavorites = jest.fn();
    const wrapper = shallow(<CardContainer movies={mockMovies} favorites={mockFavorites} checkFavorites={mockCheckFavorites} />, { disableLifecycleMethods: true });

    wrapper.find('button').simulate('click');

    expect(mockCheckFavorites).toHaveBeenCalled();
  });
});
