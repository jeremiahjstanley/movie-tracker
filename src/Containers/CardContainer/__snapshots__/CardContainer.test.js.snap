// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    checkFavorites={[Function]}
    favorites={
        Array [
            Object {
              "movie_id": 7,
              "title": "ConAir",
            },
          ]
    }
    movies={
        Array [
            Object {
              "favorite": true,
              "id": 7,
              "poster_path": "google.com",
              "title": "ConAir",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-card"
>
          <Link
                    replace={false}
                    to="/movies/ConAir"
          >
                    <div
                              className="favorite movie-image"
                              style={
                                        Object {
                                                  "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                                }
                              }
                    />
          </Link>
          <button
                    onClick={[Function]}
          >
                    Favorite
          </button>
</div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "[object Object]0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/movies/ConAir"
>
              <div
                            className="favorite movie-image"
                            style={
                                          Object {
                                                        "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                                      }
                            }
              />
</Link>,
            <button
              onClick={[Function]}
>
              Favorite
</button>,
          ],
          "className": "movie-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="favorite movie-image"
                style={
                                Object {
                                                "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                              }
                }
/>,
              "replace": false,
              "to": "/movies/ConAir",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "0 + ConAir",
              "nodeType": "host",
              "props": Object {
                "className": "favorite movie-image",
                "style": Object {
                  "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-card"
>
            <Link
                        replace={false}
                        to="/movies/ConAir"
            >
                        <div
                                    className="favorite movie-image"
                                    style={
                                                Object {
                                                            "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                                          }
                                    }
                        />
            </Link>
            <button
                        onClick={[Function]}
            >
                        Favorite
            </button>
</div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "[object Object]0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/movies/ConAir"
>
                <div
                                className="favorite movie-image"
                                style={
                                                Object {
                                                                "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                                              }
                                }
                />
</Link>,
              <button
                onClick={[Function]}
>
                Favorite
</button>,
            ],
            "className": "movie-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="favorite movie-image"
                  style={
                                    Object {
                                                      "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                                                    }
                  }
/>,
                "replace": false,
                "to": "/movies/ConAir",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "0 + ConAir",
                "nodeType": "host",
                "props": Object {
                  "className": "favorite movie-image",
                  "style": Object {
                    "backgroundImage": "url(https://image.tmdb.org/t/p/w600_and_h900_bestv2/google.com)",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
